phpunit/functional/SupplierTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2025 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace tests\units;
+
+/* Test for src/Supplier.php */
+
+class SupplierTest extends \DbTestCase
+{
+    public function testClone()
+    {
+        $this->login();
+
+        $date = date('Y-m-d H:i:s');
+        $_SESSION['glpi_currenttime'] = $date;
+
+        $this->setEntity('_test_root_entity', true);
+
+        // Create trype
+        $supplier_type = new \SupplierType();
+        $supplier__type_id = $supplier_type->add([
```

Comment:
Using $this->createItem() make code simpler and more safe. It's really better to use it.
You ensure data are created as expected (all fields), then there is no need to make an assertion after creation.

Reply:
Agree with @SebSept